.github/workflows/testing.yml

```
@@ -32,7 +32,7 @@ jobs:
       - name: Global Setup
         run: |
           python -m pip install -U pip  # Official recommended way
-          pip install pytest-xdist
+          pip install pytest-xdist==3.5.0
```

I'm not sure this is needed
```suggestion
          pip install pytest-xdist
```